index.d.ts

```diff
@@ -541,6 +541,27 @@ console.log(stdout);
 export function execaCommand(command: string, options?: Options): ExecaChildProcess;
 export function execaCommand(command: string, options?: Options<null>): ExecaChildProcess<Buffer>;
 
+type TemplateExpression = string | number | Array<string | number>;
+
+type Execa$<EncodingType = string> = {
+	(options: Options): Execa$;
+	(options: Options<null>): Execa$<null>;
+	(
+		templates: TemplateStringsArray,
+		...expressions: TemplateExpression[]
+	): EncodingType extends null
+		? ExecaChildProcess<Buffer>
+		: ExecaChildProcess;
+	sync(
+		templates: TemplateStringsArray,
+		...expressions: TemplateExpression[]
+	): EncodingType extends null
+		? ExecaSyncReturnValue<Buffer>
+		: ExecaSyncReturnValue;
+};
```

Comment:
I believe the conditionals might be simplified by using directly:
 
```ts
type Execa$<StdoutErrorType = string> = {
	(options: Options): Execa$<StdoutErrorType>;
	(options: Options<null>): Execa$<Buffer>;
	(
		templates: TemplateStringsArray,
		...expressions: TemplateExpression[]
	): ExecaChildProcess<StdoutErrorType>;
	sync(
		templates: TemplateStringsArray,
		...expressions: TemplateExpression[]
	): ExecaSyncReturnValue<StdoutErrorType>;
};
```

What do you think?

Reply:
It just keeps getting better with each iteration! I'll update to this and add the above mentioned test cases 👌

Reply:
Actually, I'm seeing a mix of `StdoutStderrorType` and `StdoutErrorType` arguments thorought the file. I have a slight preference for the more explicit `StdoutStderrorType` variant. Mind if I standardize all references?

Reply:
Yes, absolutely, good catch :+1:

Reply:
I had to make one small adjustment, but your suggestions worked great! https://github.com/sindresorhus/execa/pull/510/commits/989f6bc1fa5f007106a9cc5a24dfd1e0e27ac033

> Yes, absolutely, good catch 👍

Done https://github.com/sindresorhus/execa/pull/510/commits/f40759b2cfeb6eeaf9fb5e7e302a655e3bb5c8fe